ios-target-app/Home/Views/HomeViewController.swift

```diff
@@ -6,13 +6,107 @@
 //
 
 import UIKit
+import MapKit
+import Combine
 
-class HomeViewController: UIViewController {
+class HomeViewController: UIViewController, MKMapViewDelegate {
+  
+  private lazy var mapView: MKMapView = {
+    let map = MKMapView()
+    map.overrideUserInterfaceStyle = .dark
+    map.isZoomEnabled = true
+    map.isScrollEnabled = true
+    return map
+  }()
+  
+  private let locationManager: LocationManager
+  private var subscribers = Set<AnyCancellable>()
+  
+  init(locationManager: LocationManager) {
+    self.locationManager = locationManager
+    super.init(nibName: nil, bundle: nil)
+    
+    setupSubscribers()
+  }
+  
+  @available(*, unavailable)
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
   
   override func viewDidLoad() {
     super.viewDidLoad()
     
-    // TODO: remove temporary implementation of background color
-    view.backgroundColor = .lightGray
+    setDefaultScreenConfigs()
+    setNavigationBar()
+    setMapConstraints()
+  }
+  
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    navigationController?.setNavigationBarHidden(false, animated: animated)
+    mapView.delegate = self
+  }
+  
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    mapView.showsUserLocation = true
+  }
+  
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    mapView.showsUserLocation = false
```

Comment:
what is the intention for this to be set on the viewWillAppear and viewWillDisappear?

Reply:
For now is ok then, later you will use a custom map annotation for the user location.